joblib/func_inspect.py

```diff
@@ -128,10 +128,20 @@ def get_func_name(func, resolv_alias=True, win_characters=True):
             # mangling of full path to filename
             parts = filename.split(os.sep)
             if parts[-1].startswith('<ipython-input'):
-                # function is defined in an IPython session. The filename
-                # will change with every new kernel instance. This hack
```

Comment:
BTW, I believe the comment is incorrect: the filename doesn't change with new kernel instances. It changes because of `N` (see new comments)